src/Connections/GitlabRepo.ts

```diff
@@ -158,6 +177,48 @@ export class GitLabRepoConnection extends CommandConnection {
         }
     }
 
+    public static getProvisionerDetails(botUserId: string) {
+        return {
+            service: "gitlab",
+            eventType: GitLabRepoConnection.CanonicalEventType,
+            type: "GitLabRepo",
+            // TODO: Add ability to configure the bot per connnection type.
+            botUserId: botUserId,
+        }
+    }
+
+    public static async getConnectionTargets(userId: string, tokenStore: UserTokenStore, config: BridgeConfigGitLab, filters: GitLabTargetFilter = {}): Promise<GitLabRepoConnectionTarget[]> {
+        // Search for all repos under the user's control.
+
+        if (!filters.instance) {
+            const results: GitLabRepoConnectionInstanceTarget[] = [];
+            for (const [name, instance] of Object.entries(config.instances)) {
+                const client = await tokenStore.getGitLabForUser(userId, instance.url);
+                if (client) {
+                    results.push({
+                        name,
+                    } as GitLabRepoConnectionInstanceTarget);
+                }
+            }
+            return results;
+        }
+        // If we have an instance, search under it.
+        const instanceUrl = config.instances[filters.instance]?.url;
+        const client = instanceUrl && await tokenStore.getGitLabForUser(userId, instanceUrl);
+        if (!client) {
+            throw new ApiError('Instance is not known or you do not have access to it.', ErrCode.NotFound);
+        }
+        const after = filters.after === undefined ? undefined : parseInt(filters.after, 10);
```

Comment:
Aside from `??` also replacing `null` this seems to be what `??` was introduced for.
```suggestion
        const after = filters.after ?? parseInt(filters.after, 10);
```

Reply:
I don't think so? `filters.after ?? foo` would return `foo` if the value *is* undefined/null rather than the opposite as we're doing here. I think I meant to use &&.